### PR TITLE
fix(test): two engine reds from fleet churn — and stop the shellout guard breaking on line drift

### PR DESCRIPTION
**Pull request by gsxdsm**
Both failures are fleet-churn fallout on `main`, not defects in the conversions. Engine goes **5 failed / 3 files → 3 failed / 1 file**; the remaining 3 are `executor-prompt`'s pause-guard question, documented in #2747.

## 1. The shellout guard was coupled to line numbers — third time

Its match key was `file:LINE:primitive:signature`, so **any edit above an audited call site** broke it while the call itself was untouched. Recent fleet conversions shifted `executor.ts` and `self-healing.ts`, and 5 sites drifted at once.

**This is the third time it has gone red this way, and the third hand re-pin.** A guard that fails on edits it does not care about trains people to re-pin it without reading it — which is exactly how a real new shellout slips through in the same commit as a drift fix.

So this removes the coupling rather than updating the numbers again. Identity is now **file + primitive + signature**, with a **per-key count**; `line` stays as documentation.

The count preserves what `line` was actually buying: a *second identical* shellout in the same file is still unmatched, because the allowlist declares how many of that exact call it audits. What is deliberately given up is distinguishing "the audited call moved" from "it stayed put" — which this guard has no reason to care about.

**Measured, all three directions:**

| mutation | result |
|---|---|
| add a NEW, different shellout | **2 failed** / 1 passed |
| **duplicate** an already-audited shellout | **2 failed** / 1 passed — what `line` used to catch |
| pure line drift above an audited call | **3 passed** — previously the false failure |

## 2. A test that predicted its own flip

`executor-execution-policy-renamed-columns` asserted `moveTask` was never called. It now rehomes the card to `inbox`.

That is not a surprise — **the test's own comment called it**:

> *"the resume router's log says 'moved back to todo' and its already-there check is another `"todo"` literal — one of the 20 sites in this method left to U5's executor slice. It is why the card stays put here rather than being rehomed to `inbox`."*

A fleet PR converted that literal, and the router now rehomes to `inbox` — the declared intake column of `noHoldIr`, the workflow under test. **The prediction landing is the evidence the conversion is right**, so the case asserts the rehome instead of the absence of a move, and additionally pins that the card is never moved to the `todo` this workflow does not declare.

What the case owns is unchanged: the dispatch-loop gate did not claim the card (no *"executor recovery preserved"* log), and the run reached a real classifier rather than falling off the end.

## Verification

`pnpm test:gate` **726**, `pnpm lint` clean, engine `tsc --noEmit` clean.

## Note on PR count

This is my fourth open PR against the one-per-worker rule, opened because it clears **red on main** — the stated priority-one exception. My other three (#2753, #2747, #2743) are rebased onto current main, green, with zero unresolved threads, waiting only on CI. I am opening nothing further until they land.

### CODE REVIEW

File: packages/engine/src/__tests__/engine-no-blocking-shellout.test.ts [hunk 1]
```diff
@@ -104,15 +104,45 @@ function scanEngineSource(): ShelloutSite[] {
...
+
   const unmatched = sites.filter((site) => {
-    const key = `${site.file}:${site.line}:${site.primitive}:${site.signature}`;
-    if (!remaining.has(key)) return true;
-    remaining.delete(key);
+    const remainingForKey = budget.get(keyOf(site)) ?? 0;
+    if (remainingForKey === 0) return true;
+    budget.set(keyOf(site), remainingForKey - 1);
     return false;
   });
```
[the remark this answers is not in the document]
Comment (gsxdsm) [reply packages/engine/src/__tests__/engine-no-blocking-shellout.test.ts:139]: Correct, and it is the standard I have been applying to other people's PRs all session — so thank you for catching it in mine.

I verified both invariants by **mutation** while writing the decoupling, which proves nothing once the mutation is reverted. Nothing in the repo would have stopped a later edit restoring line coupling or weakening duplicate detection.

Fixed in `421c296649`. `classifySites` now takes the allowlist as an optional parameter (defaulting to the real one) so the semantics can be exercised against a fixture rather than against live source, and four focused cases pin them:

| case | invariant |
|---|---|
| accepts an audited call that has **moved** | line drift alone is not a violation |
| rejects a **second identical** shellout | the count is what line numbers were buying |
| reports an entry with no call site as **stale** | the other half of the budget walk |
| still distinguishes a **different** shellout in the same file | signature identity is not file-wide |

Each mutation now fails the case that owns it, plus the live guard:

| mutation | result |
|---|---|
| restore line coupling in the key | **2 failed** — the drift case and the real allowlist scan |
| weaken the budget to unbounded | **3 failed** — drift, duplicate, and the real scan |
| as shipped | **7 passed** |

On your point that the extra-call test used a different signature: that is exactly right, and it is why the duplicate case needed its own fixture — an extra call with a *different* signature exercises key-miss, not count exhaustion. They are separate failure modes and now have separate cases.

Gate 726, lint clean, engine tsc clean.

**Comment by gsxdsm** [comment]
## Added: the third engine red — a workflow with NO wip lane swallowed graph failures

`68381db72f` fixes the regression I reported on #2703. It was still red on main and I had offered there first; rather than leave a fail-closed invariant silent, I took it — folded into this PR since it already touches that exact test file, keeping to one open PR.

### The invariant that broke

`executor-execution-policy-renamed-columns` → *"a workflow with no wip column terminalizes visibly instead of claiming the card advanced"* returned `status: null, error: null`. The card was neither terminalized nor visibly failed. That test names itself *"the fail-closed direction that matters"*.

### Mechanism

`resolveResumeLanes` defaulted `wip: lifecycle?.wip ?? "in-progress"`, collapsing two different states:

1. **the IR failed to resolve** — defaulting is right; the `catch` arm wants exactly this
2. **the IR resolved and declares NO wip column** — defaulting *invents a lane the workflow does not have*

`routeGraphFailureToExecutionResume` then admitted a card resting in that workflow's **hold** lane with incomplete steps (`incompleteSteps && live.column === lanes.hold`), rehomed it, and returned `true` — so the terminalize branch below never ran. Resuming execution into a workflow with no implementation lane *is* "claiming the card advanced" when nothing did.

### Fix

`resolveResumeLanes` now also reports **`wipDeclared`** (declared, as opposed to defaulted), and the resume router declines when it is false.

This is the same fail-closed rule the sibling branch in the same method already applies with `wipColumn !== undefined` before calling a card "already advanced" — **that path failed closed while this one failed open.**

IR-unavailable deliberately keeps today's behaviour: the `catch` arm reports `wipDeclared: true`, so an infrastructure error does not start refusing legitimate resumes.

| check | result |
|---|---|
| remove the guard | **1 failed** — the fail-closed case goes red again |
| always decline | **1 failed** — a legitimate resume breaks |
| as shipped | **31 passed** across the three affected files |

### One correction against my own first guess

Three `toEqual` shape assertions needed the new field — that is the API change surfacing, not a behaviour change. I initially wrote `wipDeclared: false` for the *"falls back to the legacy trio when no workflow resolves"* case and it failed. **"No workflow resolves" does not mean "no columns"**: `resolveWorkflowIrForTask` falls back to the DEFAULT coding lineage, which declares `in-progress`. So `wipDeclared` is true there and a resume is legitimate — the case this guard exists for is an IR that resolves and declares no wip lane at all.

### Blast radius

`executor-abort-provenance`'s 2 failures are **byte-identical to main's baseline** — verified by diffing the failure sets before and after, not assumed. `executor-prompt`'s 3 are the pause-guard question documented in #2747. Gate 726, lint clean, engine tsc clean.

This PR now clears all three engine reds on main: the line-coupled shellout guard, the `todo`-literal dispatch gate, and this one.

**Comment by gsxdsm** [comment]
## Full-suite blast radius: zero new failures

Promised in the previous comment, now measured across the whole engine package rather than the three affected files:

```
Test Files  2 failed | 831 passed | 1 skipped (834)
     Tests  5 failed | 10825 passed | 14 skipped | 1 todo (10845)
```

Both failing files are the known pre-existing sets:

| file | failures | status |
|---|---:|---|
| `executor-prompt` | 3 | the pause-guard question documented in #2747 — reachable, severity-bounded, awaiting a layer decision |
| `executor-abort-provenance` | 2 | byte-identical to main's baseline, verified by diffing the failure sets |

**Engine goes 8 → 5 on main**, and the three cleared are exactly this PR's:

1. the line-coupled shellout allowlist (drifted on unrelated `executor.ts` edits, third occurrence)
2. the `todo`-literal dispatch gate (flipped exactly as its own comment predicted)
3. the no-wip-lane resume router swallowing graph failures

Nothing else in 10,845 tests moved. Given this touches lifecycle routing, that was the number worth having before asking anyone to merge it — a targeted pass on three files would not have shown a regression somewhere else in the executor.
